comictaggerlib/taggerwindow.py

```diff
@@ -1580,26 +1634,32 @@ def copy_tags(self) -> None:
                 failed_list = []
                 success_count = 0
                 for prog_idx, ca in enumerate(ca_list, 1):
-                    QtCore.QCoreApplication.processEvents()
-                    if prog_dialog.wasCanceled():
-                        break
-
-                    prog_dialog.setValue(prog_idx)
-                    prog_dialog.setLabelText(str(ca.path))
-                    QtCore.QCoreApplication.processEvents()
-
-                    if ca.has_metadata(src_style) and ca.is_writable():
-                        md = ca.read_metadata(src_style)
-
-                        if dest_style == "cbi" and self.config[0].Comic_Book_Lover__apply_transform_on_bulk_operation:
+                    ca_saved = False
+                    for style in dest_styles:
+                        if ca.has_metadata(style):
+                            QtCore.QCoreApplication.processEvents()
+                            if prog_dialog.wasCanceled():
+                                break
+
+                            prog_dialog.setValue(prog_idx)
+                            prog_dialog.setLabelText(str(ca.path))
+                            center_window_on_parent(prog_dialog)
+                            QtCore.QCoreApplication.processEvents()
+
+                        if ca.has_metadata(src_style) and ca.is_writable():
+                            md = ca.read_metadata(src_style)
+
+                        if style == "cbi" and self.config[0].Comic_Book_Lover__apply_transform_on_bulk_operation:
                             md = CBLTransformer(md, self.config[0]).apply()
 
-                        if not ca.write_metadata(md, dest_style):
-                            failed_list.append(ca.path)
-                        else:
-                            success_count += 1
+                            if not ca.write_metadata(md, style):
+                                failed_list.append(ca.path)
+                            else:
+                                if not ca_saved:
+                                    success_count += 1
+                                    ca_saved = True
 
-                        ca.load_cache([self.load_data_style, self.save_data_style, src_style, dest_style])
+                        ca.load_cache([self.load_data_style, self.save_data_style, src_style, style])
```

Comment:
this needs a de-dent, reset the cache before loading and the cache only needs to include `self.load_data_style` and `*self.save_data_styles`